exporter.go

```diff
@@ -62,11 +63,44 @@ func getEnvDuration(key string, defaultValue time.Duration) time.Duration {
 	return d
 }
 
-func createClientWithRetries(getClient func() (interface{}, error), retries int, retryInterval time.Duration) (interface{}, error) {
+// positiveDuration is a wrapper of time.Duration to ensure only positive values are accepted
+type positiveDuration struct{ duration *time.Duration }
+
+func (pd positiveDuration) String() string {
+	return fmt.Sprint(pd.duration)
+}
+
+func (pd positiveDuration) Set(s string) error {
```

Comment:
currently, if you run `./nginx-prometheus-exporter -nginx.scrape-uri=http://demo.nginx.com/stub_status -nginx.timeout=3s -nginx.retry-interval=10s`, you don't set either the timeout or the interval, because the receiver here is not a pointer 
```suggestion
func (pd *positiveDuration) Set(s string) error {
```